.github/workflows/android.yaml

```diff
@@ -46,3 +50,55 @@ jobs:
     with:
       platform: android-arm
       nightly: ${{ github.event.inputs.nightly }}
+      keep_artifacts: cobalt.apk
+
+  final_android_job:
+    runs-on: ubuntu-latest
+    needs: [ android-arm, android-arm64, android-x86 ]
+    permissions:
+      actions: write
+    steps:
+      - name: Download arm-gold apk
+        uses: actions/download-artifact@v4
+        with:
+          name: android-arm-gold
+          path: arm-gold
+      - name: Download arm-qa apk
+        uses: actions/download-artifact@v4
+        with:
+          name: android-arm-qa
+          path: arm-qa
+      - name: Download arm64-gold apk
+        uses: actions/download-artifact@v4
+        with:
+          name: android-arm64-gold
+          path: arm64-gold
+      - name: Download arm64-qa apk
+        uses: actions/download-artifact@v4
+        with:
+          name: android-arm64-gold
+          path: arm64-qa
+      - name: Download x86-gold apk
+        uses: actions/download-artifact@v4
+        with:
+          name: android-x86-gold
+          path: x86-gold
+      - name: Download x86-qa apk
+        uses: actions/download-artifact@v4
+        with:
+          name: android-x86-gold
+          path: x86-qa
+      - name: 'Upload Android APKs'
+        uses: actions/upload-artifact@v4
+        with:
+          name: Android APKs
+          path: ./*
+          retention-days: 90
+          compression-level: 0 # We expect kept artifacts to be already compressed
+          if-no-files-found: error
+      - name: 'Delete intermediates'
+        uses: geekyeggo/delete-artifact@9d15d164b1dcd538ff1b1a2984bc2c0240986c3b # v4.0.0
+        with:
+          token: ${{ github.token }}
+          name: android-arm-qa
```

Comment:
Why are you uploading and then deleting `android-arm-qa`?

Reply:
I was intending to delete all intermediates actually, forgot. Unsure if this action actually works properly with artifact v4 though

Reply:
I'm getting an `Resource not accessible by integration` error when this runs.

The doc says:
```
Support for actions/upload-artifact@v4 utilizes the GitHub REST API, and requires a permissive [GITHUB_TOKEN](https://docs.github.com/en/actions/security-guides/automatic-token-authentication#permissions-for-the-github_token), or a PAT with read and write access to actions.
``` 

I thought we do have a permissive token already